### PR TITLE
Add node 6, 10 and 14 to GitHub Actions and Azure Pipeline tests

### PR DESCRIPTION
**Pull request by kuleshovilya**
Add node 6, 10 and 14 to GitHub Actions and Azure Pipeline tests, lower npm requirements in make.js due to it being unable to use node 6

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,8 +1,18 @@
+strategy:
+  matrix:
+    'Node 8': 
+      versionSpec: '8.x'
```
Comment (anatolybolshakov) [inline azure-pipelines.yml:4]: Please add tests for node 6 also - since currently it is still used as execution handler

**Review by anatolybolshakov — APPROVED** [review]

**Review by DaniilShmelev — COMMENTED** [review]
Just curious - why do we have both github actions and azure pipelines CI? As I can see, github actions runs `npm run build` and `npm test`, and azure pipelines runs `npm run units`. Should we merge them/get rid of one of them?
I think it would be easiest to just add `npm run units` command to github actions and remove azure pipelines.
@anatolybolshakov what do you think?

**Review by alexander-smolyakov — APPROVED** [review]

**Comment by alexander-smolyakov** [comment]
> Just curious - why do we have both github actions and azure pipelines CI? As I can see, github actions runs `npm run build` and `npm test`, and azure pipelines runs `npm run units`. Should we merge them/get rid of one of them?
> I think it would be easiest to just add `npm run units` command to github actions and remove azure pipelines.
> @anatolybolshakov what do you think?

@DaniilShmelev, the azure pipeline also publishes the build artifact that we could use later to publish the package to npm.
https://github.com/microsoft/typed-rest-client/blob/8bdfb968c1207ce755685124c2139a7febf7cbad/azure-pipelines.yml#L38-L44

Overall I agree with the idea to merge these pipelines into one and if we will decide to do this let's save the pipeline in Azure.

**Comment by kuleshovilya** [comment]
@alexander-smolyakov @DaniilShmelev So, what are we deciding on? Merging both tasks, as in adding unit tests and publish build to a github actions and removing azure pipelines, while saving it to azure repo?

**Comment by alexander-smolyakov** [comment]
@kuleshovilya I would like to suggest the following approach:
1) Merge this PR
2) Update related Azure pipeline and remove GitHub action
3) Open another PR

**Review by DaniilShmelev — APPROVED** [review]

**Review by anatolybolshakov — APPROVED** [review]

**Pull request merged**